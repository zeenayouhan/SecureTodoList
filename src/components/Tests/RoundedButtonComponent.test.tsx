import React from 'react';
import { render, fireEvent } from '@testing-library/react-native';
import RoundedButton from '../RoundedButton.component';

describe('RoundedButton', () => {
  const onPressMock = jest.fn();

  it('renders the button with the given label', () => {
    const { getByText } = render(
      <RoundedButton label='Test Button' onPress={onPressMock} />
    );
    expect(getByText('Test Button')).toBeTruthy();
  });

  it('calls the onPress function when the button is pressed', () => {
    const { getByText } = render(
      <RoundedButton label='Test Button' onPress={onPressMock} />
    );
    const button = getByText('Test Button');
    fireEvent.press(button);
    expect(onPressMock).toHaveBeenCalled();
  });
});
describe('Rendering', () => {
  it('should match to snapshot', () => {
    const onPressMock = jest.fn();
    const roundedButtonComponent = render(
      <RoundedButton label='Test Button' onPress={onPressMock} />
    );
    expect(roundedButtonComponent).toMatchSnapshot();
  });
});
