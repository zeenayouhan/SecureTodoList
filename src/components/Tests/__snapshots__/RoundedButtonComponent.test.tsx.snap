// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rendering should match to snapshot 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": undefined,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#182387",
      "borderRadius": 60,
      "flexDirection": "row",
      "height": 110,
      "justifyContent": "center",
      "marginVertical": 20,
      "opacity": 1,
      "padding": 24,
      "paddingVertical": 32,
      "width": 300,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#FFFFFF",
          "fontSize": 18,
          "fontWeight": "bold",
        },
        undefined,
      ]
    }
  >
    Test Button
  </Text>
</View>
`;
